source/Ship.cpp

```diff
@@ -2719,6 +2719,10 @@ int Ship::Scan(const PlayerInfo &player)
 	if(isYours || (target->isYours && activeScanning))
 		Audio::Play(Audio::Get("scan"), Position());
 
+	bool isImportant = false;
+	if(target->isYours)
+		isImportant = government->FinesContents(target.get()) || target.get() == player.Flagship();
```

Comment:
Left operand is evaluated first and the pointer comparison is going to be way faster than the other thing.
```suggestion
		isImportant = target.get() == player.Flagship() || government->FinesContents(target.get());
```